src/mpid/ch3/channels/nemesis/include/mpid_nem_defs.h

```diff
@@ -42,13 +41,14 @@ extern  char *MPID_nem_asymm_base_addr;
 
 static inline MPID_nem_cell_ptr_t MPID_NEM_REL_TO_ABS (MPID_nem_cell_rel_ptr_t r)
 {
-    return (MPID_nem_cell_ptr_t)((char*)MPL_atomic_relaxed_load_ptr(&r.p) + (MPID_nem_addr_t)MPID_nem_asymm_base_addr);
+    void *p = MPL_atomic_relaxed_load_ptr(&r.p);
+    return (MPID_nem_cell_ptr_t) (void *) ((uintptr_t) p + (uintptr_t) MPID_nem_asymm_base_addr);
```

Comment:
By this comment https://github.com/pmodels/mpich/pull/5283/files#r636462145, if you mean to remove the `uintptr_t` cast, then no. You are not allowed to add two pointers. If you retain the `uintptr_t` cast, then I think we would prefer some consistency, i.e. do the same cast in both place since the semantics are the same.

Reply:
I guess replace the first `uintptr_t` cast with `char *` is okay.